generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts

```diff
@@ -26,12 +27,12 @@ import { ActiveMenuDirective } from './layouts/navbar/active-menu.directive';
         <%=angular2AppName%>AccountModule
     ],
     declarations: [
+        <%=jhiPrefixCapitalized%>LoginModalComponent,
```

Comment:
no just declare `LoginModalComponent` and then export it from the `common.ng2module` as below, now import the `<%=angular2AppName%>CommonModule` in the `account.ng2module` and you should be able to access any exported members from there. home and navbar are not reused hence they can be in the `app.ng2module`
```
@NgModule({
    imports: [
        <%=angular2AppName%>SharedModule
    ],
    declarations: [
        jhiAlertComponent,
        PageRibbonComponent,
        <%=jhiPrefixCapitalized%>LoginModalComponent
    ],
    providers: [
        AlertService,
        ProfileService,<%_ if (enableTranslation){ _%>
        <%=jhiPrefixCapitalized%>LanguageService,
        <%_ } _%>
        Account,
        Principal,
        AuthService
    ],
    exports: [
        <%=jhiPrefixCapitalized%>LoginModalComponent
    ],
    schemas: [CUSTOM_ELEMENTS_SCHEMA]

})
export class <%=angular2AppName%>CommonModule {}
```